cilium/network_policy.h

```diff
@@ -123,12 +122,9 @@ class PortPolicy : public Logger::Loggable<Logger::Id::config> {
   bool forRange(std::function<bool(const PortNetworkPolicyRules&, bool& denied)> allowed) const;
   bool forFirstRange(std::function<bool(const PortNetworkPolicyRules&)> f) const;
 
-  static bool hasHttpRules(const PolicyMap& map, const RulesList& wildcard_rules,
-                           const PolicyMap::const_iterator port_rules);
-
   const PolicyMap& map_;
-  const RulesList& wildcard_rules_;
-  const PolicyMap::const_iterator port_rules_; // iterator to 'map_'
+  const PortNetworkPolicyRules* wildcard_rules_;
```

Comment:
why not use smart pointer here and below?

Reply:
So that we don't accidentally keep (last) reference to the policy object in a worker thread. When worker threads request the policy, they get it by reference, which is never stored across the function calls, so there are never dangling references.

Reply:
Added comment to state the above.